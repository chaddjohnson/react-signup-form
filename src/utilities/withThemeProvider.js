import React from 'react';
import { ThemeProvider } from 'emotion-theming';
import theme from '../theme';

const withThemeProvider = Component => props => (
  <ThemeProvider theme={theme}>
    <Component {...props} />
  </ThemeProvider>
);

export default withThemeProvider;
