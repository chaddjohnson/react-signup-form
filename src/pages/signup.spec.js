import React from 'react';
import renderer from 'react-test-renderer';
import SignupPage from './signup';
import withThemeProvider from '../utilities/withThemeProvider';

const ThemedSignupPage = withThemeProvider(SignupPage);

describe('SignupPage', () => {
  it('should render', () => {
    const SignupPageComponent = renderer.create(<ThemedSignupPage />).toJSON();
    expect(SignupPageComponent).toMatchSnapshot();
  });
});
