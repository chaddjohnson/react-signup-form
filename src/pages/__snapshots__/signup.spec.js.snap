// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupPage should render 1`] = `
<div
  className="css-1r9g6c5"
>
  <div
    className="css-1fbbzml"
  >
    <header
      className="css-c9spwi"
    >
      <h2>
        Sign Up
      </h2>
    </header>
    <form
      onSubmit={[Function]}
    >
      <div
        className="css-12lkxey"
      >
        <label
          className="css-4rywna"
          htmlFor="username"
        >
          Username
        </label>
        <input
          autoFocus={true}
          className="css-3w2per"
          id="username"
          maxLength={32}
          minLength={6}
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Choose a username"
          type="text"
          value=""
        />
      </div>
      <div
        className="css-12lkxey"
      >
        <label
          className="css-4rywna"
          htmlFor="password"
        >
          Password
        </label>
        <input
          autoFocus={false}
          className="css-3w2per"
          id="password"
          minLength={8}
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Enter a strong password"
          type="password"
          value=""
        />
      </div>
      <div
        className="css-12lkxey"
      >
        <label
          className="css-4rywna"
          htmlFor="passwordConfirmation"
        >
          Confirm Password
        </label>
        <input
          autoFocus={false}
          className="css-3w2per"
          id="passwordConfirmation"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Confirm your password"
          type="password"
          value=""
        />
      </div>
      <div
        className="css-12lkxey"
      >
        <footer
          className="css-1tclyyl"
        >
          <button
            className="css-1xtj9u"
            disabled={true}
            onClick={[Function]}
          >
            Sign Up
          </button>
        </footer>
      </div>
    </form>
  </div>
</div>
`;
